Source/Core/Core/HW/WiimoteEmu/MotionPlus.cpp

```diff
@@ -67,118 +87,170 @@ ExtensionPort& MotionPlus::GetExtPort()
-      {
-        // Deactivate motion plus:
-        reg_data.ext_identifier[2] = INACTIVE_DEVICE_ADDR << 1;
-        reg_data.cert_ready = 0x0;
+      // Passthrough to the connected extension. (if any)
+      return m_i2c_bus.BusWrite(slave_addr, addr, count, data_in);
+    }
 
-        // Pass through the activation write to the attached extension:
-        // The M+ deactivation signal is cleverly the same as EXT activation:
-        i2c_bus.BusWrite(slave_addr, addr, count, data_in);
-      }
-      // TODO: kill magic number
-      else if (0xf1 == addr)
-      {
-        INFO_LOG(WIIMOTE, "M+ cert activation: 0x%x", reg_data.cert_enable);
-        // 0x14,0x18 is also a valid value
-        // 0x1a is final value
-        reg_data.cert_ready = 0x18;
-      }
-      // TODO: kill magic number
-      else if (0xf2 == addr)
-      {
-        INFO_LOG(WIIMOTE, "M+ calibration ?? : 0x%x", reg_data.unknown_0xf2[0]);
-      }
+    auto const result = RawWrite(&m_reg_data, addr, count, data_in);
 
-      return result;
-    }
-    else
+    if (PASSTHROUGH_MODE_OFFSET == addr)
     {
-      // No i2c passthrough when activated.
-      return 0;
+      OnPassthroughModeWrite();
     }
+
+    return result;
   }
-  else
+
+  case ActivationStatus::Active:
   {
-    if (INACTIVE_DEVICE_ADDR == slave_addr)
+    // FYI: Motion plus does not respond to 0x53 when activated.
+    if (ACTIVE_DEVICE_ADDR != slave_addr)
     {
-      auto const result = RawWrite(&reg_data, addr, count, data_in);
-
-      // It seems a write of any value triggers activation.
-      if (0xfe == addr)
-      {
-        INFO_LOG(WIIMOTE, "M+ has been activated: %d", data_in[0]);
+      // No i2c passthrough when activated.
+      return 0;
+    }
 
-        // Activate motion plus:
-        reg_data.ext_identifier[2] = ACTIVE_DEVICE_ADDR << 1;
-        // TODO: kill magic number
-        // reg_data.cert_ready = 0x2;
+    auto const result = RawWrite(&m_reg_data, addr, count, data_in);
 
-        // A real M+ is unresponsive on the bus for some time during activation
-        // Reads fail to ack and ext data gets filled with 0xff for a frame or two
-        // I don't think we need to emulate that.
+    if (offsetof(Register, initialized) == addr)
+    {
+      // It seems a write of any value here triggers deactivation on a real M+.
+      Deactivate();
 
-        // TODO: activate extension and disable encrption
-        // also do this if an extension is attached after activation.
-        std::array<u8, 1> data = {0x55};
-        i2c_bus.BusWrite(ACTIVE_DEVICE_ADDR, 0xf0, (int)data.size(), data.data());
+      // Passthrough the write to the attached extension.
+      // The M+ deactivation signal is cleverly the same as EXT activation.
+      m_i2c_bus.BusWrite(slave_addr, addr, count, data_in);
+    }
+    else if (offsetof(Register, init_stage) == addr)
+    {
+      if (m_reg_data.init_stage == 0x01)
+      {
+        m_reg_data.init_progress = 0x18;
```

Comment:
Are those `0x01` and `0x18` magic numbers important enough for a named constant (or at least a comment)?
Stage looks like it could be 1, 2, 3, ... in ascending order, and progress...a percentage perhaps? If so, of what?

Reply:
I have comments at the definition of `init_progress` in MotionPlus.h. `init_stage` is only ever 0 or 1. But I can add a comment.